hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/AclManager.java

```diff
@@ -0,0 +1,402 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.hadoop.ipc.ProtobufRpcEngine;
+import org.apache.hadoop.ipc.Server;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+import org.apache.hadoop.ozone.om.helpers.OmVolumeArgs;
+import org.apache.hadoop.ozone.security.acl.IAccessAuthorizer;
+import org.apache.hadoop.ozone.security.acl.OzoneObj;
+import org.apache.hadoop.ozone.security.acl.OzoneObjInfo;
+import org.apache.hadoop.ozone.security.acl.RequestContext;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+
+import static org.apache.hadoop.ozone.om.OzoneManager.getS3Auth;
+import static org.apache.hadoop.ozone.om.exceptions.OMException.ResultCodes.INVALID_REQUEST;
+import static org.apache.hadoop.ozone.om.lock.OzoneManagerLock.Resource.BUCKET_LOCK;
+import static org.apache.hadoop.ozone.om.lock.OzoneManagerLock.Resource.VOLUME_LOCK;
+
+/**
+ * AclManager is used to manage the ACLs for Ozone resources.
+ */
+public class AclManager {
+  public static final Logger LOG =
+      LoggerFactory.getLogger(AclManager.class);
+
+  private final OMMetadataManager metadataManager;
+  private final InetSocketAddress omRpcAddress;
+  private final IAccessAuthorizer accessAuthorizer;
+  private final boolean isNativeAuthorizerEnabled;
+
+  public AclManager(OMMetadataManager metadataManager,
+                    InetSocketAddress omRpcAddress,
+                    IAccessAuthorizer accessAuthorizer,
+                    boolean isNativeAuthorizerEnabled) {
+    this.metadataManager = metadataManager;
+    this.omRpcAddress = omRpcAddress;
+    this.accessAuthorizer = accessAuthorizer;
+    this.isNativeAuthorizerEnabled = isNativeAuthorizerEnabled;
+  }
+
+  public void checkAcls(OzoneObj.ResourceType resType,
+                        OzoneObj.StoreType store,
+                        IAccessAuthorizer.ACLType acl,
+                        String vol,
+                        String bucket,
+                        String key)
+      throws IOException {
+    UserGroupInformation user;
+    if (getS3Auth() != null) {
+      String principal =
+          OzoneAclUtils.accessIdToUserPrincipal(getS3Auth().getAccessId());
+      user = UserGroupInformation.createRemoteUser(principal);
+    } else {
+      user = ProtobufRpcEngine.Server.getRemoteUser();
+    }
+
+    InetAddress remoteIp = ProtobufRpcEngine.Server.getRemoteIp();
+
+    checkAllAcls(resType, store, acl,
+        vol, bucket, key,
+        user != null ? user : getRemoteUser(),
+        remoteIp != null ? remoteIp : omRpcAddress.getAddress(),
+        remoteIp != null ? remoteIp.getHostName()
+            : omRpcAddress.getHostName());
+  }
+
+  @SuppressWarnings("parameternumber")
```

Comment:
I think these annotations to suppresswarnings for parameter number are unnecessary and can be removed from the refactored acls.  Ran a build without, checkstyle and findbugs, did not report a problem.